From 04c047f8d9d1cc5c725df0fed57cdc6ccce54d7f Mon Sep 17 00:00:00 2001
From: Hans de Goede <johannes.goede@oss.qualcomm.com>
Date: Mon, 27 Jul 2026 16:00:32 +0200
Subject: [PATCH 30/39] pipeline: rpi: Explictly pass pipe()->name() when
 creating IPA

Switch to explicitly specifying the pipeline-name as the name of the IPA
to create.

This is a preparation patch for switching the default IPA creation to
matching IPA by IPA proxy type, which does not work for rpi because
the same IPAProxyRPi type is used for both the rpi/vc4 and rpi/pisp
pipeline handlers.

Signed-off-by: Hans de Goede <johannes.goede@oss.qualcomm.com>
(cherry picked from commit 68e9564ca2a20abcab05e546093cc273afaa75a2)
---
 src/libcamera/pipeline/rpi/common/pipeline_base.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libcamera/pipeline/rpi/common/pipeline_base.cpp b/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
index 5a5acf6a1..d1f4bc79f 100644
--- a/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
+++ b/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
@@ -1165,7 +1165,7 @@ int CameraData::loadIPA(ipa::RPi::InitResult *result)
 {
 	int ret;
 
-	ipa_ = pipe()->createIPA<ipa::RPi::IPAProxyRPi>(1, 1);
+	ipa_ = pipe()->createIPA<ipa::RPi::IPAProxyRPi>(pipe()->name(), 1, 1);
 
 	if (!ipa_)
 		return -ENOENT;
